### ISSUE TITLE
Build issue - unable to create flexflow_cffi_header.py

### ISSUE THREAD

**Issue by santhnm2** [flexflow/FlexFlow#251]
I have been unable to build FlexFlow due to what appears to be an issue creating flexflow_cffi_header.py. Here's the relevant snippet from my build log (the full log is here: [ff_build.log](https://github.com/flexflow/FlexFlow/files/7841773/ff_build.log)):
```
Creating flexflow_cffi_header.py...
cd /future/u/keshav2/flexflow/python && /future/u/keshav2/flexflow/python/flexflow_cffi_build.py --ffhome-dir /future/u/keshav2/flexflow --libname flexflow_pybind11_internal.cpython-39-x86_64-linux-gnu.so --output-dir /future/u/keshav2/flexflow/python/flexflow/core
cd /future/u/keshav2/flexflow/build/python && /lfs/1/keshav2/miniconda3/envs/flexflow/lib/python3.9/site-packages/cmake/data/bin/cmake -E cmake_link_script CMakeFiles/flexflow_python.dir/link.txt --verbose=1
/usr/bin/c++ -O3 -DNDEBUG -UNDEBUG -rdynamic CMakeFiles/flexflow_python.dir/main.cc.o -o ../../python/flexflow_python  -Wl,-rpath,/future/u/keshav2/flexflow/build:/future/u/keshav2/flexflow/build/python:/usr/local/cuda/lib64/stubs:/usr/local/cuda/lib64:/future/u/keshav2/flexflow/build/deps/legion/lib: -lrt -ldl -rdynamic -Wl,--whole-archive ../libflexflow.so -Wl,--no-whole-archive libflexflow_c.so -lz /usr/local/cuda/lib64/stubs/libcuda.so /usr/local/cuda/lib64/libcublas.so /usr/local/cuda/lib64/libcurand.so /usr/local/cuda/lib64/libcudnn.so -lrt -ldl -rdynamic ../deps/legion/lib/liblegion.so.1 ../deps/legion/lib/librealm.so.1
../libflexflow.so: undefined reference to `void Reshape::forward_kernel<float>(float const*, float*, unsigned long, CUstream_st*)'
collect2: error: ld returned 1 exit status
make[2]: *** [python/CMakeFiles/flexflow_python.dir/build.make:110: ../python/flexflow_python] Error 1
make[2]: Leaving directory '/future/u/keshav2/flexflow/build'
make[1]: *** [CMakeFiles/Makefile2:337: python/CMakeFiles/flexflow_python.dir/all] Error 2
make[1]: Leaving directory '/future/u/keshav2/flexflow/build'
make: *** [Makefile:136: all] Error 2
```

I ran the same build command with the same repository (shared on a distributed file system) on a different server and it was able to successfully build there, so I'm not sure what's different about this environment. The only difference I can identify is the compiler version - on the broken server, the compiler is:
```
-- The C compiler identification is GNU 5.4.0
-- The CXX compiler identification is GNU 5.4.0
```
On the working server the compiler is: 
```
-- The C compiler identification is GNU 9.3.0
-- The CXX compiler identification is GNU 9.3.0
```
Both of these are using the default system compiler, but if I install a newer compiler through the conda environment on the broken server I encounter the issue described in issue #175. Is this likely to be a compiler version issue or is there something else going wrong?

**Comment by jiazhihao** [comment]
It seems you used Makefile to build FlexFlow. Can you switch to cmake (https://github.com/flexflow/FlexFlow/blob/master/INSTALL.md#2-build-the-flexflow)? We recommend users to use cmake to compile FlexFlow.

**Comment by santhnm2** [comment]
I followed the instructions at that link which were:
```
mkdir build
cd build
../config/config.linux
make
```
So the output I included is from the `make` command after running `config/config.linux` which I believe calls `run_cmake`. Is there an alternate way of compiling with cmake you recommend?

**Comment by jiazhihao** [comment]
The issue was a link error 
```
../libflexflow.so: undefined reference to `void Reshape::forward_kernel<float>(float const*, float*, unsigned long, CUstream_st*)' 
```

Can you check if the issue is fixed by the most recent commit `9930e7f06105`?

**Comment by santhnm2** [comment]
Yup that fixed it, thanks Zhihao!
